Update errors , fix panic if db was not configured

msarajam opened the pull request:
# Description

Adding if statement to check if db is nil , this is fixing the panic in testcases and also if one of the function related to db called

Organizing the error messages in app.go and grpc,go 
*/TODO/* 
the error organization needs a little more work so all the errors are following the same format
and/or be more centralized


Fixes # (issue)

Fix Panic in testcases caused by db==nil

## Type of change

Please delete options that are not relevant.
- [x] Bug fix (non-breaking change which fixes an issue)

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, ensuring GoDoc readability and clarity in hard-to-understand areas
- [x] I have made corresponding changes to documentation
- [-] I have added tests that ensure my fix is effective or that my feature works
- [-] Any dependent changes have been merged and published in downstream modules

If checklist items are unchecked please explain.

adding test was not needed
changing dependencies was not needed

Commit:
Update errors , fix panic if db was not configured

msarajam commented:
Changes are just in 6 files (including http.go and http_test.go they are empty and are just for reminder of these needs to worked on)
rest of the files are vendors and mods

msarajam committed:
Merge branch 'develop' into Momo

madflojo commented on databases/cassandra/cassandra_test.go:
```diff
@@ -66,7 +70,7 @@ func TestHappyPath(t *testing.T) {
 		err := db.Set("test_happypath", data)
 		if err != nil {
-			t.Errorf("Unexpected error when writing data - %s", err)
+			t.Fatalf("Unexpected error when writing data - %s", err)
```
👍 didn't know about `t.Fatal()`

msarajam replied:
👍

madflojo committed:
Delete http.go

Let's wait to add these when we have something to add. I'd rather not having lingering files with nothing in them.

madflojo committed:
Delete http_test.go

Same with this

madflojo commented on app/app.go:
```diff
@@ -27,6 +29,17 @@ var db databases.Database
+
+// This part is for constant
+var (
+	cassandraName = "casandra"
```
Do we really need this when we only use it once?

msarajam replied:
not really , but if the program grows then might be easier

msarajam replied:
i removed it , we can added later if we had to

madflojo commented on databases/cassandra/cassandra.go:
```diff
@@ -111,6 +116,10 @@ func Dial(conf *Config) (*Database, error) {
 // already been initialized this function will not execute but return with a nil error. If any issues occur
 // while initializing an error will be returned.
 func (db *Database) Initialize() error {
+	//
+	if db == nil {
+		return fmt.Errorf("%s , db = %v", dbNill, db)
+	}
```
I'm trying to understand how we could call `Initialize()` part of the DB interface with a nil db?

msarajam replied:
i am not sure why , but it was getting nil when i was running the test cases ,

it was not happening if i was running the application

madflojo replied:
Interesting, but that sounds more like a problem in the tests area than in the DB `Initialize()` function. Let's add our nil checks there.

madflojo commented on databases/cassandra/cassandra.go:
```diff
@@ -130,6 +139,11 @@ func (db *Database) Initialize() error {
+	if err != nil {
+		return fmt.Errorf("Unable to initialize data base - %s ", err)
+	}
+
```
We wouldn't want this right? Because there is an error check on line 126 which checks if it's any error other than `ErrNoKeyspace`. If it is `ErrNoKeyspace` then we created it and all is happy now.

msarajam replied:
you are right , i saw that it is && so i added that . i'll remove it

madflojo replied:
Cool, thanks!

msarajam replied:
i removed it , Done

madflojo commented on databases/cassandra/cassandra_test.go:
```diff
@@ -58,6 +58,10 @@ func TestHappyPath(t *testing.T) {
+	if db == nil {
+		t.Fatalf("Database has not been configured , db = t %v", db)
+	}
+
```
Maybe we should make line 58 a `t.Fatalf` too. That would cover this scenario and be a bit less code. But good catch either way.

msarajam replied:
i'll make that change ,

the t.Error used to failed the test case , i guess it is in new release

madflojo replied:
Yeah, `t.Error` will only fail the test case but `t.Fatal` will stop executing the tests. It's cool and easy.

msarajam replied:
i made all the changes ,

madflojo commented on app/app.go:
```diff
@@ -62,7 +75,7 @@ func Run(cfg *config.Config) error {
 	err := db.Initialize()
 	if err != nil {
-		return fmt.Errorf("Unable to initilize the database - %s", err)
+		return fmt.Errorf("%s - %s", unableInitDB, err)
```
Same question here, do we need this to be a variable when it's only used once?

msarajam replied:
i tried to have all the errors together , that was all , but there is no other benefit to have it separate

madflojo commented on app/app.go:
```diff
@@ -49,11 +62,11 @@ func Run(cfg *config.Config) error {
 		db, err = cassandra.Dial(Config.Databases.Cassandra)
 		if err != nil {
-			return fmt.Errorf("Unable to connect to cassandra database - %s", err)
+			return fmt.Errorf("%s %s- %s", unableConnectDB, cassandraName, err)
```
Also here, same question.

msarajam replied:
same thing :D

msarajam replied:
but also if we add more DBs then we can use this in separate function and call that

madflojo replied:
let's skip adding these for now, maybe they will be useful in the future but right now I'm not sure they are needed.

msarajam replied:
ok

madflojo committed:
Update grpc.go

madflojo committed:
Update grpc.go

You might need to run `gofmt` on this again after merging this.

madflojo committed:
Gofmt this thing

msarajam committed:
changing errof to fatalf in cassandra_test.go

msarajam committed:
changing the name of the erros (golint)

msarajam committed:
Merge branch 'Momo' into momo

msarajam committed:
Merge pull request #1 from madflojo/momo

Gofmt this thing

msarajam committed:
init http commit

msarajam committed:
init http , update vendor

msarajam committed:
Merge branch 'Momo' into http

msarajam committed:
Merge pull request #3 from msarajam/http

Http init commit

msarajam committed:
fix vendor

msarajam committed:
test with travis,yml

msarajam committed:
test with travis,yml

msarajam committed:
test with travis,yml

msarajam committed:
test with travis,yml

madflojo commented on app/app.go:
```diff
@@ -7,12 +7,14 @@ package app
 	"strings"
 	"time"
+
+	"hord/config"
```
You'll need to specify the full path or this won't build. When you clone this make sure you clone it into `github.com/madflojo/hord` that way the full path is in your `$GOPATH`

msarajam replied:
but , is it the right way to do it ?

madflojo replied:
yes?

msarajam replied:
was it a question ? :D

msarajam committed:
Listen to my God

msarajam committed:
update go mod
